google/cloud/opentelemetry/internal/monitored_resource.cc

```diff
@@ -52,7 +52,7 @@ struct AsStringVisitor {
 
 struct OTelKeyMatch {
   std::vector<std::string> otel_keys;
-  absl::optional<std::string> fallback = absl::nullopt;
+  std::string fallback = {};
```

Comment:
Ugh, can you add a `NOLINT`, and I'll clean this up on my own time in a follow up PR?

```suggestion
  // NOLINTNEXTLINE(readability-redundant-member-init)
  std::string fallback = {};
```

I have no idea why clang-tidy did not complain for the previous code. Whatever.